Dnd.Ddd.Model/Character/Saga/CharacterCreationSaga.cs

```diff
@@ -71,7 +58,22 @@ protected virtual void Dispose(bool disposing)
             }
         }
 
-        private void CheckSagaCompletion(CharacterDraft draft, BaseDomainEvent domainEvent)
+        private async Task HandleNotification(CharacterEvent notification, CancellationToken cancellationToken)
+        {
+            var character = await GetCharacter(notification, cancellationToken);
+            Guard.With<InvalidCharacterStateException>().Against(character.State is Completed, notification.CharacterUiD);
+            CheckSagaCompletion(character, notification);
+        }
+
+        private async Task<Character> GetCharacter(CharacterEvent notification, CancellationToken cancellation)
+        {
+            var character = await characterRepository.GetAsync(notification.CharacterUiD, cancellation) ?? 
```

Comment:
character variable is redundant, consider removing it

Reply:
thanks, done